### PR TITLE
Relax matching to ignore whitespace on each line

### PR DESCRIPTION
**Pull request by tautropfli**
See [#62](https://github.com/Emoun/duplicate/pull/62)

### CODE REVIEW

File: src/expand.rs [hunk 1]
```diff
@@ -265,6 +265,14 @@ fn run_tests<I, S>(
     }
 }
 
+fn matches_expected(expected: &str, actual: &str) -> bool {
+    actual.contains(expected) || trim_each_line(actual).contains(&trim_each_line(expected))
+}
+
+fn trim_each_line(input: &str) -> String {
```
Comment (Emoun) [inline src/expand.rs:275]: Collecting the lines like this means there is not more line separation. So the following two will match:

```
something1
somethign2
```
```
something1something2
```
We probably don't want that. Maybe you can re-introduce newlines alone, such that the lines are matched against each other exclusively

File: src/expand.rs [hunk 2]
```diff
@@ -265,6 +265,14 @@ fn run_tests<I, S>(
     }
 }
 
+fn matches_expected(expected: &str, actual: &str) -> bool {
```
Comment (Emoun) [inline src/expand.rs:269]: Please add doc comments. 
I know the code base doesn't have too much of it itself, but I'd like new code to improve on it.

**Review by Emoun — CHANGES_REQUESTED** [review]
Thanks for this PR. I like the idea of your solution, but I hope you can find a more robust matching. See the comments.

File: src/expand.rs (hunk 1 above)
Comment (tautropfli) [reply src/expand.rs:275]: oh you're absolutely right. I think I *may* have rushed this a bit

**Comment by tautropfli** [comment]
All done, line endings are now preserved :)

**Review by Emoun — APPROVED** [review]
Thanks for this PR and the fixes. Seems good to me now

**Pull request merged**